modvendor/modvendor_test.go

```diff
@@ -15,18 +15,823 @@
 package modvendor_test
 
 import (
+	"bytes"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
 	"path/filepath"
+	"regexp"
+	"strings"
 	"testing"
+	"text/template"
 
 	"github.com/madlambda/spells/assert"
 	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/hcl"
 	"github.com/mineiros-io/terramate/modvendor"
 	"github.com/mineiros-io/terramate/test"
+	errtest "github.com/mineiros-io/terramate/test/errors"
 	"github.com/mineiros-io/terramate/test/sandbox"
 	"github.com/mineiros-io/terramate/tf"
 	"github.com/rs/zerolog"
+
+	. "github.com/mineiros-io/terramate/test/hclwrite/hclutils"
 )
 
+// vendorPathSpec describes paths inside a vendor directory when its
+// directory is not yet know. It's format is just:
+//   <module source>#<vendor relative path>
+// Where the module source can use the "text/template" (dot) variable to be
+// substituted at a later time when the final vendor directory is computed.
+//
+// Example:
+//   git::file://{{.}}/module-test?ref=main#main.tf
+type vendorPathSpec string
+
+type testcase struct {
+	name         string
+	source       string
+	layout       []string
+	vendordir    string
+	configs      []hclconfig
+	wantVendored []string
+	wantIgnored  []wantIgnoredVendor
+	wantError    error
+
+	// wantFiles describes file contents that must exist in the vendor directory.
+	// The fmt.Stringer can be a template that gets compiled with text/template
+	// package and having access to the relative paths of the vendored modules.
+	// For example, if `wantVendored` contains [<modsrc1>, <modsrc2>, ...], then
+	// the template will have access to the list [<relative path of modsrc1>,
+	// <relative path of modsrc2>, ...].
+	// Then the template can reference the third element of the list:
+	//   source = {{index 2}}
+	// If this looks like a hack to you, it's because it is.
+	wantFiles map[vendorPathSpec]fmt.Stringer
+}
+
+type hclconfig struct {
+	repo string
+	path string
+	data fmt.Stringer
+}
+
+type wantIgnoredVendor struct {
+	RawSource     string
+	ReasonPattern string
+}
+
+type wantReport struct {
+	Vendored []string
+	Ignored  []wantIgnoredVendor
+	Error    error
+}
+
+func TestModVendorAllRecursive(t *testing.T) {
```

Comment:
```suggestion
func TestModVendor(t *testing.T) {
```

I think we only export on the package the Vendor function that is always recursive no ?